xcode/Subconscious/Shared/UIKit/BlockEditor/Blocks/ListBlockCell.swift

```diff
@@ -0,0 +1,311 @@
+//
+//  QuoteBlockCell.swift
+//  BlockEditor
+//
+//  Created by Gordon Brander on 7/20/23.
+//
+
+import UIKit
+import SwiftUI
+
+extension BlockEditor {
+    class ListBlockCell:
+        UICollectionViewCell,
+        UITextViewDelegate,
+        UIViewComponentProtocol
+    {
+        static let identifier = "ListBlockCell"
+        
+        var id: UUID = UUID()
+        
+        weak var delegate: TextBlockDelegate?
+        
+        private lazy var selectView = BlockEditor.BlockSelectView()
+        private lazy var stackView = UIStackView()
+        private var listContainerMargins = NSDirectionalEdgeInsets(
+            top: 0,
+            leading: AppTheme.unit4,
+            bottom: 0,
+            trailing: 0
+        )
+        private lazy var listContainer = UIView()
+        private lazy var textView = SubtextTextView()
+        private lazy var bulletView = createBulletView()
+        private var transcludeMargins = NSDirectionalEdgeInsets(
+            top: AppTheme.unit,
+            leading: AppTheme.padding,
+            bottom: AppTheme.padding,
+            trailing: AppTheme.padding
+        )
+        private lazy var transcludeListView = BlockEditor.TranscludeListView()
+        
+        private lazy var toolbar = UIToolbar.blockToolbar(
```

Comment:
oof

Reply:
Yeah, I went back and forth on this, and decided that it is best to have each block be stand-alone. In future we will have block types with very unique behavior, so basing the paradigm on a shared class seems like a mistake.

The verbosity comes down to how hard it is to do code sharing in UIKit.